src/subdomains/core/history/controllers/transaction.controller.ts

```diff
@@ -18,41 +8,117 @@ import {
+  constructor(
+    private readonly historyService: HistoryService,
+    private readonly userService: UserService,
+    private readonly transactionService: TransactionService,
+    private readonly buyCryptoWebhookService: BuyCryptoWebhookService,
+    private readonly buyFiatService: BuyFiatService,
+    private readonly refRewardService: RefRewardService,
+    private readonly bankDataService: BankDataService,
+    private readonly bankTxService: BankTxService,
+    private readonly fiatService: FiatService,
+  ) {}
 
   @Get()
   @ApiOkResponse({ type: TransactionDto, isArray: true })
   async getCsvCompact(
     @Query() query: HistoryQueryUser,
-    @Response({ passthrough: true }) res,
+    @Res({ passthrough: true }) res: Response,
   ): Promise<TransactionDto[] | StreamableFile> {
     if (!query.format) query.format = ExportFormat.JSON;
     return this.getHistoryData(query, ExportType.COMPACT, res);
   }
 
+  @Get('detail')
+  @ApiBearerAuth()
+  @UseGuards(AuthGuard(), new RoleGuard(UserRole.USER))
+  @ApiOkResponse({ type: TransactionDetailDto, isArray: true })
+  async getTransactionDetails(
+    @GetJwt() jwt: JwtPayload,
+    @Query() query: TransactionFilter,
+    @Res({ passthrough: true }) res: Response,
+  ): Promise<TransactionDetailDto[] | StreamableFile> {
+    const user = await this.userService.getUser(jwt.id, { userData: { users: true } });
+    const txList = await this.transactionService.getTransactionsForUsers(user.userData.users, query.from, query.to);
+
+    // map to DTO
+    const dtoList = await Util.asyncMap(txList, async (tx) => {
+      if (tx.buyCrypto) {
+        const bc = await this.buyCryptoWebhookService.extendBuyCrypto(tx.buyCrypto);
+        return TransactionDtoMapper.mapBuyCryptoTransactionDetail(bc);
+      } else if (tx.buyFiat) {
+        const bf = await this.buyFiatService.extendBuyFiat(tx.buyFiat);
+        return TransactionDtoMapper.mapBuyFiatTransactionDetail(bf);
+      } else if (tx.refReward) {
+        const rr = await this.refRewardService.extendReward(tx.refReward);
+        return TransactionDtoMapper.mapReferralRewardDetail(rr);
+      }
+
+      return undefined;
+    }).then((list) => list.filter((dto) => dto));
+
+    if (query.format === ExportFormat.CSV) {
+      this.setCsvResult(res, ExportType.COMPACT);
+      return this.historyService.getCsv(dtoList, ExportType.COMPACT);
+    }
+
+    return dtoList;
+  }
+
+  @Get('unassigned')
+  @ApiBearerAuth()
+  @UseGuards(AuthGuard(), new RoleGuard(UserRole.USER))
+  @ApiOkResponse({ type: UnassignedTransactionDto, isArray: true })
+  async getUnassignedTransactions(@GetJwt() jwt: JwtPayload): Promise<UnassignedTransactionDto[]> {
+    const user = await this.userService.getUser(jwt.id, { userData: true });
+    const bankDatas = await this.bankDataService.getBankDatasForUser(user.userData.id);
+
+    const txList = await this.bankTxService.getUnassignedBankTx(bankDatas.map((bd) => bd.iban));
```

Comment:
We will not find all bankTx with it. Theoretically we have to revert the senderAccount Method in bankTx entity and call it on the bankData iban 🤔

Reply:
Good point. For now it should only be used for transactions with real IBANs. We will need to find a solution for all transactions later.